app/models/planning_application.rb

```diff
@@ -633,6 +634,16 @@ def longitude
     super || lonlat.try(:x)
   end
 
+  def update_lonlat
+    return unless longitude.present? && latitude.present?
+
+    update!(lonlat: factory.point(longitude, latitude))
+  end
```

Comment:
Can't we do this in a `before_create` rather than running an `UPDATE` right after the `INSERT` ?